packages/react-charting/UnitTests/LineChartUT.test.tsx

```diff
@@ -0,0 +1,533 @@
+    LineChart,
+    { data: basicChartPoints, allowMultipleShapesForPoints: false, theme: DarkTheme },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(9);
+      fireEvent.mouseOver(points[1]);
+      expect(getById(container, /circle/i)[1].getAttribute('fill')).toEqual('#1b1a19');
+    },
+  );
+
+  testWithWait(
+    'Should return white if the point is active and allowMultipleShapesForPoints is true and pointIndex is 1',
+    LineChart,
+    { data: basicChartPoints, allowMultipleShapesForPoints: true, theme: DarkTheme },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(9);
+      fireEvent.mouseOver(points[1]);
+      expect(getById(container, /circle/i)[1].getAttribute('fill')).toEqual('#1b1a19');
+    },
+  );
+
+  testWithWait(
+    'Should return yellow if the point is not active and allowMultipleShapesForPoints is true and pointIndex is 1',
+    LineChart,
+    { data: basicChartPoints, allowMultipleShapesForPoints: true, theme: DarkTheme },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(9);
+      fireEvent.mouseOver(points[1]);
+      expect(getById(container, /circle/i)[0].getAttribute('fill')).toEqual('yellow');
+    },
+  );
+
+  testWithWait(
+    'Should return red if the point is not active, multiple shapes for points allowed and point is not last or second',
+    LineChart,
+    { data: basicChartPoints, allowMultipleShapesForPoints: true, theme: DarkTheme },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(9);
+      fireEvent.mouseOver(points[2]);
+      expect(getById(container, /circle/i)[1].getAttribute('fill')).toEqual('yellow');
+    },
+  );
+});
+
+runTest('Unit tests for __injectIndexPropertyInLineChartData function', () => {
+  test('Should return an empty array if lineChartData is undefined', () => {
+    const instance = new LineChartBase({ data: emptyData });
+    const result = instance._injectIndexPropertyInLineChartData(emptyData.lineChartData);
+    expect(result).toEqual([]);
+  });
+
+  test('Should generate a color for each item in lineChartData if color is not defined', () => {
+    const instance = new LineChartBase({ data: simpleChartPoints });
+    const result = instance._injectIndexPropertyInLineChartData(simpleChartPoints.lineChartData);
+    expect(result).toHaveLength(3);
+    expect(result[0].color).toBeDefined();
+    expect(result[1].color).toBeDefined();
+    expect(result[2].color).toBeDefined();
+  });
+
+  test('Should generate a color for each item in lineChartData with tokens when color is defined', () => {
+    const instance = new LineChartBase({ data: basicChartPoints });
+    const result = instance._injectIndexPropertyInLineChartData(basicChartPoints.lineChartData);
+    expect(result).toHaveLength(3);
+    expect(result[0].color).toBeDefined();
```

Comment:
check for actual value and not to be defined only?

Reply:
done